numpyro/distributions/batch_util.py

```diff
@@ -563,6 +564,9 @@ def _promote_batch_shape_expanded(d: ExpandedDistribution):
     return new_self
 
 
+promote_batch_shape.register(Independent, _promote_batch_shape_expanded)
```

Comment:
Mmh, I don't see any reason why the logic for batch shape promotion of `Independent` and `ExpandedDistribution` should be the same. Promoting `ExpandedDistribution` requires some care in maintaining broadcastability between itself and its `base_dist` when left-most inserting batch dimensions - I don't think `Independent` would require this kind of care.

Could you instead create  `_promote_batch_shape_independent` which simply promotes an independent's base distribution?

Reply:
Mmmh, unclear - `MaskedDistribution` enforces `dist.batch_shape == dist.base_dist.batch_shape` while an `Independent` distribution will have a different batch shape that its base.

Reply:
I've committed an ` _promote_batch_shape_independent` now, it passes the test, and outputs the correct batch_shape, but beyond that I'm not sure whether the logic is correct.